tags/state/transformer.js

```diff
@@ -0,0 +1,162 @@
+const esprima = require("esprima");
+const escodegen = require("escodegen");
+const estraverse = require("estraverse");
+
+function getValue(body) {
+  let lValue = body.left && escodegen.generate(body.left);
+  let argValue = body.argument && escodegen.generate(body.argument);
+  let rValue = body.right && escodegen.generate(body.right);
+
+  switch (body.operator) {
+    case "++":
+      return `++${argValue}`;
+    case "--":
+      return `--${argValue}`;
+    case "+=":
+      return `${lValue} + ${rValue}`;
+    case "-=":
+      return `${lValue} - ${rValue}`;
+    default:
+      return rValue;
+  }
+}
+
+function findExpression(ast, id) {
+  let isChanged;
+  const replace = estraverse.replace(ast, {
+    enter: function(node) {
+      if (
+        node.type === "AssignmentExpression" ||
+        node.type === "UpdateExpression"
+      ) {
+        if (
+          (node.left && node.left.name === id) ||
+          (node.argument && node.argument.name === id)
+        ) {
+          const value = getValue(node);
+          const newScript = esprima.parseScript(
+            `(component.setState("${id}", ${value}))`,
+            { range: true }
+          );
+          isChanged = true;
+          this.skip();
+          return newScript.body[0].expression;
+        }
+      }
+    }
+  });
+  return !isChanged ? null : ast;
+}
+
+function getNewAst(ast, id, builder, node) {
+  const newAst = findExpression(ast, id);
+  if (newAst) {
+    let newCode = escodegen.generate(newAst);
+    if (newCode[newCode.length - 1] === ";") {
+      newCode = newCode.slice(0, newCode.length - 1);
+    }
+    if (newCode[0] === "_") {
+      newCode = newCode.slice(1);
+    }
+    // Wrap new expression in parenthesies to prevent any statement code like semicolins to cause it to fail
+    if (node) {
+      if (node.type === "Scriptlet") {
+        return builder.scriptlet({ value: newCode });
+      }
+    }
+    const newExpression = builder.expression(newCode);
+    return newExpression;
+  }
+}
+
+function lookFor(context, id) {
+  const builder = context.builder;
+  const walker = context.createWalker({
+    enter(node) {
+      if (node.type === "HtmlElement") {
+        const properties = node.getProperties();
+
+        if (properties) {
+          Object.keys(properties).forEach(prop => {
+            if (prop.indexOf("on") === 0) {
+              const currentProp = properties[prop];
+              currentProp.args = currentProp.args.map(arg => {
+                if (arg.type === "Expression") {
+                  const newAst = getNewAst(arg.ast, id, builder);
+                  return newAst || arg;
+                }
+                return arg;
+              });
+            }
+          });
+        }
+      } else if (
+        node.type === "HtmlAttribute" &&
+        node.name.indexOf("on") === 0
```

Comment:
Instead of explicitly checking for `on` it might make more sense to just check for `.argument`